api/resources_portal/management/commands/material_additional_metadata_key_change.py

```diff
@@ -0,0 +1,183 @@
+
+
+def material_additional_metadata_key_change(category, old_key, new_key, pagination):
+    query_set = None
+    materials_changed = 0
+
+    if category:
+        query_set = Material.objects.filter(category=category)
+    else:
+        query_set = Material.objects.all()
+
+    filtered_query_set = query_set.filter(additional_metadata__has_key=old_key)
+    not_applicable_query_set = query_set.exclude(additional_metadata__has_key=old_key)
+
+    # Evaluates the queryset. Querysets are usually lazy, this forces evaluation
+    len(not_applicable_query_set)
+
+    def add_func(material):
+        material.additional_metadata[new_key] = material.additional_metadata.pop(old_key)
+        return material
+
+    materials_changed = paginate_queryset(filtered_query_set, add_func, pagination)
+
+    logger.info(f"The key '{old_key}' was changed to '{new_key}' in {materials_changed} materials.")
+
+    log_nonapplicable_materials(not_applicable_query_set, old_key)
+
+
+class Command(BaseCommand):
+    """
+    This is a command to add, remove, or change the names of additional metadata fields currently in the server.
+    It has the following use cases:
+    1. Change a metadata key
+    ./rportal migrate-metadata-keys old_key new_key --category [CATEGORY] --pagination [PAGINATION]
+    2. Add a metadata key
+    ./rportal migrate-metadata-keys --add new_key --category [CATEGORY] --default [DEFAULT] --pagination [PAGINATION]
+    The default is the the value that all the new keys will be initialized with.
+    2. Remove a metadata key
+    ./rportal migrate-metadata-keys --remove old_key --category [CATEGORY] --pagination [PAGINATION]
+    For each use case, the category parameter is optional, and categories should be specified in all caps, like "PLASMID" or "MODEL_ORGANISM".
+    The pagination parameter is also optional. It controls the size of the queryset pagination.
+    """
+
+    help = "Replaces all instances of old_key in materials (of an optional category) with new_key."
+
+    def add_arguments(self, parser):
+        # Positional arguments
+        parser.add_argument("old_key", nargs="?", default=None, type=str)
+        parser.add_argument("new_key", nargs="?", default=None, type=str)
+
+        # Optional arguments
+        parser.add_argument("--category", nargs="?", default=None)
+        parser.add_argument("--pagination", nargs="?", default=None, type=int)
+        parser.add_argument("--add", nargs="?", default=None)
+        parser.add_argument("--default", nargs="?", default="")
+        parser.add_argument("--remove", nargs="?", default=None)
+
+    def handle(self, *args, **options):
+        category = options["category"]
+        if category is not None:
+            if not validate_category(category):
+                raise CommandError(
+                    f"The provided category '{category}' is not a valid material category."
+                )
+
+        # Default query pagination is 100
+        pagination = 100
```

Comment:
This is actually on the low side, I think a page of 1000 would be fine!

Reply:
Changed to 1000!